### PR TITLE
feat: add transcript audio command

### PR DESCRIPTION
**Pull request by mauvieira**
## Audio Transcription using OpenAI

Transcription of quoted audio messages in WhatsApp using the Whisper API from OpenAI. The transcription is supported for various audio formats and is integrated into the existing middleware architecture.

## Usage
The transcription process is triggered when a user quotes an audio message using the command `!transcript`
The bot downloads the quoted audio, saves it, transcribes it using the Whisper API, and replies with the transcribed text.

### CODE REVIEW

**Review by akinncar — APPROVED** [review]

File: src/wweb/onMessage/audio/transcriptAudioMessage.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const transcriptAudioMessage = async (msg: Message): Promise<void> => {
+  if (!msg.hasQuotedMsg) {
+    return;
+  }
+
+  const quotedMsg = await msg.getQuotedMessage();
+  handleQuotedAudioMessage(quotedMsg, msg);
+};
+
+export default middleware(transcriptAudioMessage);
```
Comment (akinncar) [inline src/wweb/onMessage/audio/transcriptAudioMessage.ts:79]: I would like to separate concepts right here if possible: isolate functions per file, and only declare `transcriptAudioMessage` in this file as we do with other functions, so `transcriptAudioMessage` threat the message and call the action, and the action is in another place to be used more generically for other things

File: src/wweb/onMessage/audio/transcriptAudioMessage.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    console.error(`Error during transcription (attempt ${retries + 1}):`, error);
+
+    if (retries < MAX_RETRIES) {
+      console.log(`Retrying transcription... (${retries + 1}/${MAX_RETRIES})`);
+      await new Promise(resolve => setTimeout(resolve, 2000));
+      return transcribeAudio(filePath, retries + 1, msg);
+    }
+    console.error('Failed to transcribe audio after multiple attempts:', error);
+  }
+};
```
Comment (vinibgoulart) [inline src/wweb/onMessage/audio/transcriptAudioMessage.ts:45]: move to a function inside openai package

File: src/wweb/onMessage/audio/transcriptAudioMessage.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    mkdirSync(dir);
+  }
+  return dir;
+};
+
+const saveMediaToFile = (media: MessageMedia, dir: string, extension: string): string => {
+  const filePath = join(dir, `${uuidv4()}.${extension}`);
+  writeFileSync(filePath, media.data, { encoding: 'base64' });
+  return filePath;
+};
```
Comment (vinibgoulart) [inline src/wweb/onMessage/audio/transcriptAudioMessage.ts:24]: Could this be in a separate file?

**Review by vinibgoulart — COMMENTED** [review]
amazing!

File: src/wweb/onMessage/audio/transcriptAudioMessage.ts (hunk 2 above)
Comment (vinibgoulart) [reply src/wweb/onMessage/audio/transcriptAudioMessage.ts:45]: suggestion: `openaiTranscriptionsCreate.ts`

**Review by vinibgoulart — APPROVED** [review]

**Review by anddreluis2 — APPROVED** [review]

**Pull request merged**